test/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/RepositoryCloneManagerTests.cs

```diff
@@ -176,6 +178,10 @@ void ResetCalls()
 
         // A third clone with a new remote
         ResetCalls();
+        // We want to make it so we don't find all requested refs in the first remote
```

Comment:
I think I prefer having these smaller `ReturnsAsync` setups rather than having a big one, since it's easier to read what happens, but I'm open to other opinions